graphql_compiler/tests/test_compiler.py

```diff
@@ -1345,7 +1356,36 @@ def test_complex_optional_variables(self):
                 )
             ])}
         '''
-        expected_sql = NotImplementedError
+        expected_sql = '''
+            SELECT
+                [FeedingEvent_1].event_date AS child_fed_at,
+                [FeedingEvent_2].event_date AS grandparent_fed_at,
+                [FeedingEvent_3].event_date AS other_parent_fed_at
+            FROM
+                db_1.schema_1.[Animal] AS [Animal_1]
+                JOIN db_1.schema_1.[Animal] AS [Animal_2]
+                    ON [Animal_1].parent = [Animal_2].uuid
+                LEFT OUTER JOIN db_2.schema_1.[FeedingEvent] AS [FeedingEvent_1]
+                    ON [Animal_2].fed_at = [FeedingEvent_1].uuid
+                JOIN db_1.schema_1.[Animal] AS [Animal_3]
+                    ON [Animal_2].uuid = [Animal_3].parent
+                LEFT OUTER JOIN db_2.schema_1.[FeedingEvent] AS [FeedingEvent_3]
+                    ON [Animal_3].fed_at = [FeedingEvent_3].uuid
+                JOIN db_1.schema_1.[Animal] AS [Animal_4]
+                    ON [Animal_1].uuid = [Animal_4].parent
+                JOIN db_2.schema_1.[FeedingEvent] AS [FeedingEvent_2]
+                    ON [Animal_4].fed_at = [FeedingEvent_2].uuid
+            WHERE (
+                ([FeedingEvent_1].uuid IS NOT NULL) = 0 OR
```

Comment:
We should be able to lower `([FeedingEvent_1].uuid IS NOT NULL) = 0` into `[FeedingEvent_1].uuid IS NULL`, right?

Reply:
The way to do this is to lower `ContextFieldExistence(V)` into `BinaryExpression('!=', V.came_from, None)`. The `came_from` value is not known during lowering. It's possible to compute it, but it's not trivial, and maybe even not worth considering this is purely a cosmetic fix.